clang/lib/AST/TextNodeDumper.cpp

```diff
@@ -677,13 +677,18 @@ void TextNodeDumper::dumpBareType(QualType T, bool Desugar) {
   ColorScope Color(OS, ShowColors, TypeColor);
 
   SplitQualType T_split = T.split();
-  OS << "'" << QualType::getAsString(T_split, PrintPolicy) << "'";
+  std::string T_str = QualType::getAsString(T_split, PrintPolicy);
+  OS << "'" << T_str << "'";
 
   if (Desugar && !T.isNull()) {
-    // If the type is sugared, also dump a (shallow) desugared type.
+    // If the type is sugared, also dump a (shallow) desugared type when
+    // visibly different.
```

Comment:
```suggestion
    // If the type is sugared, also dump a (shallow) desugared type when
    // it is visibly different.
```

Reply:
What's wrong with what's there? It's grammatically correct AFAIK.